extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/HttpServerCommonHandlers.java

```diff
@@ -143,6 +147,27 @@ public void handle(RoutingContext event) {
         }
     }
 
+    private static void addFilterHeaders(RoutingContext event, Map<String, String> headers) {
+        for (var entry : headers.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            MultiMap responseHeaders = event.response().headers();
+            List<String> oldValues = responseHeaders.getAll(key);
+            if (oldValues.isEmpty()) {
+                responseHeaders.set(key, value);
+            } else {
+                // we need to make sure the new value is not duplicated
+                var newValues = new LinkedHashSet<String>(oldValues);
+                boolean added = newValues.add(value);
+                if (added) {
+                    responseHeaders.set(key, newValues);
+                } else {
+                    // we don't need to do anything here as the value was already in the set
```

Comment:
Remove the unnecessary `else` block then?
```suggestion
                } else {
                    // we don't need to do anything here as the value was already in the set
```

Reply:
I see, but it seems to be explicit in your code already and the IDE nags about empty else blocks 😜

Reply:
I generally disable that inspection 😉